src/Cardano/Wallet.hs

```diff
@@ -135,11 +183,32 @@ mkWalletLayer db network = WalletLayer
                 throwE $ ErrCreateWalletIdAlreadyExists wid
     , readWallet = \wid -> liftIO (readCheckpoint db (PrimaryKey wid)) >>= \case
```

Comment:
Would you need to move this into `readWalletId/readWallet` below?

Reply:
👍 Ah yes. Should definitely re-use the helper here.